platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -131,13 +135,31 @@ private void doRollback(SolrClient solrClient, String type) {
   }
 
   @Override
-  // Returned Map will have suffixes in the key names - client is responsible for handling them
+  /**
+   * {@inheritDoc} Returned Map will have suffixes in the key names - client is responsible for
+   * handling them
+   */
   public List<Map<String, Object>> get(String type, String cql) throws PersistenceException {
+    return get(type, cql, DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE);
+  }
+
+  @Override
+  public List<Map<String, Object>> get(String type, String cql, int startIndex, int pageSize)
+      throws PersistenceException {
     if (StringUtils.isBlank(type)) {
       throw new PersistenceException(
           "The type of object(s) to retrieve must be non-null and not blank, e.g., notification, metacard, etc.");
     }
 
+    if (startIndex < 0) {
+      throw new PersistenceException("The start index must be positive.");
```

Comment:
Are these really `PersistenceException`s or should they be `IllegalArgumentException`s?

Reply:
Fixed.